plan/find_best_task.go

```diff
@@ -252,6 +252,14 @@ func (ds *DataSource) findBestTask(prop *requiredProp) (t task, err error) {
 	t = invalidTask
 
 	for _, path := range ds.possibleAccessPaths {
```

Comment:
Maybe it's better to remove other access paths if there is a path which has an empty range. This can be done in [`DataSource.deriveStats()`](https://github.com/pingcap/tidb/blob/master/plan/stats.go#L138)

Reply:
updated